Sources/MapboxMaps/Foundation/RetroactiveConformances/DispatchQueueProtocol.swift

```diff
@@ -0,0 +1,27 @@
+import Dispatch
+
+// depending on this protocol instead of on DispatchQueue directly
+// allow mocking the main queue in tests which avoids the need for waits
+internal protocol DispatchQueueProtocol: AnyObject {
+    func async(
+        group: DispatchGroup?,
+        qos: DispatchQoS,
+        flags: DispatchWorkItemFlags,
+        execute work: @escaping @convention(block) () -> Void
+    )
+    func async(execute workItem: DispatchWorkItem)
+}
+
+extension DispatchQueueProtocol {
+    func async(
+        group: DispatchGroup? = nil,
+        qos: DispatchQoS = .unspecified,
+        flags: DispatchWorkItemFlags = [],
+        execute work: @escaping @convention(block) () -> Void
+    ) {
+        async(group: group, qos: qos, flags: flags, execute: work)
+    }
```

Comment:
I have some concerns about this one.
Methods implementation in protocol extensions has static dispatching if the method was not included in the protocol. 
IIRC function with default values would be represented by method without args.
I think we need to discuss this approach and do some extra testing.